public/app/plugins/panel/nodeGraph/NodeGraphPanel.tsx

```diff
@@ -5,6 +5,7 @@ import { NodeGraph } from '@grafana/ui';
 import { useLinks } from '../../../features/explore/utils/links';
 
 export const NodeGraphPanel: React.FunctionComponent<PanelProps<Options>> = ({ width, height, data }) => {
+  const getLinks = useLinks(data.timeRange);
   if (!data || !data.series.length) {
```

Comment:
Why this change? Isn't it redundant no run useLinks if we pass no data and we are not going to use the getLinks at the end?

Reply:
While it makes sense what you are saying it violates the [rules of hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level).

> Don’t call Hooks inside loops, conditions, or nested functions. Instead, always use Hooks at the top level of your React function. By following this rule, you ensure that Hooks are called in the same order each time a component renders. That’s what allows React to correctly preserve the state of Hooks between multiple useState and useEffect calls.